control/velocity_controller/CMakeLists.txt

```diff
@@ -1,77 +1,23 @@
-cmake_minimum_required(VERSION 3.0.2)
+cmake_minimum_required(VERSION 3.5)
 project(velocity_controller)
 
-add_compile_options(-std=c++14)
+find_package(ament_cmake_auto REQUIRED)
+ament_auto_find_build_dependencies()
 
-find_package(Eigen3 REQUIRED)
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
 
-find_package(
-  catkin REQUIRED COMPONENTS
-  roscpp
-  tf2
-  tf2_ros
-  geometry_msgs
-  autoware_planning_msgs
-  autoware_control_msgs
-  dynamic_reconfigure
-)
-
-generate_dynamic_reconfigure_options(
-  cfg/VelocityController.cfg
-)
-
-catkin_package(
-  INCLUDE_DIRS include
-  CATKIN_DEPENDS
-    roscpp
-    tf2
-    tf2_ros
-    geometry_msgs
-    autoware_planning_msgs
-    autoware_control_msgs
-)
-
-###########
-## Build ##
-###########
-
-include_directories(
-  include
-  ${catkin_INCLUDE_DIRS}
-)
 
-add_executable(
-  velocity_controller
+ament_auto_add_executable(velocity_controller
   src/velocity_controller_core.cpp
   src/velocity_controller_node.cpp
```

Comment:
please also add the headers. This will help IDEs to discover which source file are part of the project. Won't change the executable that's produced

Reply:
Do you mean, to write like this?
```
ament_auto_add_executable(velocity_controller
  include/velocity_controller/delay_compensation.h
  include/velocity_controller/lowpass_filter.h
  include/velocity_controller/pid.h
  include/velocity_controller/velocity_controller_mathutils.h
  include/velocity_controller/velocity_controller.h
  src/velocity_controller_core.cpp
  src/velocity_controller_node.cpp
  src/pid.cpp
  src/velocity_controller_mathutils.cpp
)
```

In cmake, we usually do like
```
include_directories(include)
```
and, I'm not familiar with the ament, but I thought the ament automatically do include_directories like above, and the headers are not needed to add here.

To "help IDEs to discover which source file are part of the project" is the function of the ament? Or your IDE?

Reply:
> In cmake, we usually do like
> include_directories(include)

That's to inform the compiler  where to look for files and becomes `c++ -Iinclude/`

In contrast, what I meant is that adding the header files as sources actually tells cmake (and any IDE that queries cmake) that these files are part of this target `velocity_controller` and thus parf of this cmake project and not just files that happen to be in a folder. See e.g. this quote from https://crascit.com/2016/01/31/enhanced-source-file-handling-with-target_sources/

> In the above example, note that .h header files were specified as sources too, not just the .cpp implementation files. Headers listed as sources don’t get compiled directly on their own, but the effect of adding them is for the benefit of IDE generators like Visual Studio, Xcode, Qt Creator, etc. This causes those headers to be listed in the project’s file list within the IDE, even if no source file refers to it via #include. This can make those headers easier to find during development and potentially aid things like refactoring functionality, etc.

Reply:
> Do you mean, to write like this?

yes, that's what I mean

Reply:
Thank you! I modified the Cmakelist. Please check it.